src/taco-utils/npmHelper.ts

```diff
@@ -0,0 +1,89 @@
+/**
+﻿ *******************************************************
+﻿ *                                                     *
+﻿ *   Copyright (C) Microsoft. All rights reserved.     *
+﻿ *                                                     *
+﻿ *******************************************************
+﻿ */
+
+/// <reference path="../typings/node.d.ts" />
+/// <reference path="../typings/npm.d.ts" />
+/// <reference path="../typings/q.d.ts" />
+
+"use strict";
+
+import npm = require("npm");
+import path = require("path");
+import Q = require("q");
+import nopt = require("nopt");
+import url = require("url");
+import semver = require("semver");
+
+var npmconf = require("npm/lib/config/core.js");
+var configDefs = npmconf.defs;
+var types = configDefs.types;
+var shorthands = configDefs.shorthands;
+
+import installLogLevel = require("./installLogLevel");
+import InstallLogLevel = installLogLevel.InstallLogLevel;
+
+module TacoUtility {
+	export class NpmHelper {
+		private static runNpmCommand(npmCommand: string, args: string[], workingDirectory: string, commandFlags?: string[], logLevel?: InstallLogLevel, optionalArgs?: any): Q.Promise<any> {
```

Comment:
Yes, view uses it to suppress console output. Some npm commands use more parameters than their signature would suggest, and this is a way to make use of that.